### ISSUE TITLE
Don't recompile unchanged files

### ISSUE THREAD

**Issue by 2016rshah** [2016rshah/heckle#10]
This isn't an issue for most blogs, but perhaps if you have thousands and thousands of posts, the process of compiling every single markdown file into HTML even if it hasn't changed won't be very efficient. Perhaps look into a way to check the diff for a file or something to only recompile changed files when you run `heckle build` to make the program faster

**Comment by arvind-iyer** [comment]
Probably not the best way but wouldn't saving the timestamp of the last `heckle build` make it possible? We can then use Posix.Files to get the time a file was last modified and only compile the ones that were modified after the last build

**Comment by 2016rshah** [comment]
Yeah but we would need to save the timestamp for the last time someone ran `heckle build` in order to do so. 

I've been considering adding a a configuration file to keep track of this and other user preferences (like [Markdown/LaTeX/both](https://github.com/2016rshah/heckle/issues/9)) but I'm hesitant to add another file for things that aren't that important. Which is more valuable, simplicity or configurability/features?
